zeppelin-interpreter/src/main/java/org/apache/zeppelin/interpreter/Interpreter.java

```diff
@@ -203,6 +204,71 @@ public void setClassloaderUrls(URL[] classloaderUrls) {
     this.classloaderUrls = classloaderUrls;
   }
 
+  /**
+   * General function to register hook event
+   * @param noteId - Note to bind hook to
+   * @param event The type of event to hook to (pre_exec, post_exec)
+   * @param cmd The code to be executed by the interpreter on given event
+   */
+  @Experimental
+  public void registerHook(String noteId, String event, String cmd) {
```

Comment:
I kind of see your point, moving the convenience methods from `Interpreter.java` to `InterpreterGroup.java` would save the need for defining wrappers in `LazyOpenInterpreter.java`. However this would also mean that the interpreter `className` must always be specified. This means for example that this line:
```Java
String cmdDev = interpreter.getHook(noteId, HookType.PRE_EXEC_DEV);
```
Would be changed to
```Java
String className = interpreter.getClassName();
String cmdDev = interpreter.getInterpreterGroup().getHook(noteId, className, HookType.PRE_EXEC_DEV);
```
Additionally while these methods are only currently called through `RemoteInterpreterServer`, I will have use cases for directly calling them in the Python `Interpreter` subclasses (eg, `PySparkInterpreter` and `PythonInterpreter`) for my upcoming matplotlib integration work. Given this, I felt that the current syntax would be clearer.